exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/statements/api/PreparedStatementApi.kt

```diff
@@ -18,10 +21,29 @@ interface PreparedStatementApi {
 
     fun addBatch()
 
+    /**
+     * Executes an SQL query stored in a [PreparedStatement].
+     *
+     * @return The [ResultSet] generated by the query.
+     */
     fun executeQuery(): ResultSet
 
+    /**
+     * Executes an SQL statement stored in a [PreparedStatement].
+     *
+     * @return The affected row count if the executed statement is a DML type;
+     * otherwise, 0 if the statement returns nothing.
+     */
     fun executeUpdate(): Int
 
+    /**
+     * Executes multiple SQL statements stored in a single [PreparedStatement].
+     *
+     * @return A list of [StatementResult]s retrieved from the database, which may store either affected row counts
+     * or [ResultSet]s. The order of elements is based on the order of the statements in the `PreparedStatement`.
+     */
+    fun executeMultiple(): List<StatementResult>
+
```

Comment:
Introducing `sealed class StatementResult` isn't strictly necessary for this PR, but having the return type of `executeMultiple()` be a `List<StatementResult>` instead of just a `ResultSet` gives flexibility for future feature extension.

It allows, for example, a version of `exec()` to be introduced (like `execMultiple()` or `execMixed()`) that returns a collection of varying results from concatenated statements, which is a feature most other ORMs already provide.